acceptance/acceptance_test.go

```diff
@@ -812,6 +811,75 @@ func testAcceptance(
 						})
 					})
 
+					when("--trust-builder", func() {
+						when("platform API version is up to 0.2", func() {
+							it.Before(func() {
+								h.SkipIf(t,
+									semver.MustParse(lifecycleDescriptor.API.PlatformVersion.String()).GreaterThan(semver.MustParse("0.2")),
+									"skipping trust-builder (and the creator). Lifecycle does not support it",
+								)
+							})
+
+							it("calls the 5 phases", func() {
+								output := h.Run(t, subjectPack(
+									"build",
+									repoName,
+									"-p",
+									filepath.Join("testdata", "mock_app"),
+								))
+								h.AssertNotContains(t, output, "CREATING")
+								h.AssertContains(t, output, "DETECTING")
+								h.AssertContains(t, output, "ANALYZING")
+								h.AssertContains(t, output, "RESTORING")
+								h.AssertContains(t, output, "BUILDING")
+								h.AssertContains(t, output, "EXPORTING")
+							})
+						})
+
+						when("platform API version is 0.3+", func() {
+							it.Before(func() {
+								h.SkipIf(t,
+									semver.MustParse(lifecycleDescriptor.API.PlatformVersion.String()).LessThan(semver.MustParse("0.3")),
+									"ignore trust-builder (and the creator). Lifecycle does not support it",
+								)
+								h.SkipIf(t,
+									!packSupports(packPath, "build --trust-builder"), "trust-builder not supported",
+								)
+							})
+
+							when("the builder is trusted", func() {
+								it("calls the creator", func() {
```

Comment:
I wonder how far we want to take this - we're asserting that we've called the creator but we don't test that a build completes successfully. It might be nice to see the simplest happy path `pack build` at a minimum. But if something were to go wrong it might be because some combination of user inputs was not correctly translated into arguments passed to the creator - which is presumably why we have phases_test.go, but perhaps acceptance tests would give us additional security that would be helpful. Thinking about how the current acceptance tests could be extended to use creator vs. not-creator (such as with a [table test](https://github.com/buildpacks/pack/blob/6bccf6c9cb0b3f5be910d1a748bb4ca8a4393bb5/internal/buildpackage/builder_test.go#L66-L78)) boggles the mind a little bit and I am not sure that we want to go down that path - I am just raising the question.

Reply:
I was thinking about this a little bit more, given that we're moving toward everything using the creator (in the short term) we could change test expectations that use the binary name (such as [this one](https://github.com/buildpacks/pack/blob/0501b34a9be65832f2b8a60cbcffbb4d1342b333/acceptance/acceptance_test.go#L683)) to be regexes. We'll still need to answer the question of trusted vs. untrusted builders and the test matrix that might introduce at some point, but it's maybe not as immediate now.

Reply:
Following the discussion in [this slack thread](https://buildpacks.slack.com/archives/CSFE49VGR/p1588091913054100?thread_ts=1587752732.046400&cid=CSFE49VGR), I'll make this changes after removing the `--trusted-builder` flag and updating the acceptance-tests accordingly.
Thanks for the idea!